be/src/storage/persistent_index.cpp

```diff
@@ -646,6 +646,7 @@ class FixedMutableIndex : public MutableIndex {
             const auto value = values[idx];
             uint64_t hash = FixedKeyHash<KeySize>()(key);
             if (auto [it, inserted] = _map.emplace_with_hash(hash, key, value); inserted) {
+                old_values[idx] = NullIndexValue;
```

Comment:
I think the old_values is initialized as NullIndexValue before call this function.

Reply:
But it will be wrong if caller missing set old_values with `NullIndexValue` before call this function, and the function will be harder to use.

Reply:
> But it will be wrong if caller missing set old_values with `NullIndexValue` before call this function, and the function will be harder to use.

Set old_values to NullIndexValue is ok,  what i means is this maybe not the cause of the estimated incorrect page num.